src/components/RichText.tsx

```diff
@@ -2,62 +2,87 @@
+} from './Typography';
 
 export type RichTextProps = {
   wysiwyg: StoryblokRichtext;
-  isDark?: boolean;
+  isLightText?: boolean;
+  textAlign?: TextAlignType;
   className?: string;
   linkColor?: string;
 };
 
 export const RichText = ({
   wysiwyg,
-  isDark,
+  isLightText,
+  textAlign,
   className,
 }: RichTextProps) => {
-  let textColor = 'su-text-current';
+  const textColor = isLightText ? 'su-text-white print:su-text-black' : 'su-text-black';
 
-  if (isDark) {
-    textColor = 'su-text-black-20 print:su-text-black';
-  }
   const rendered = render(wysiwyg, {
     markResolvers: {
-      styled: (children, props) => React.createElement(
-        'span',
-        {
-          // eslint-disable-next-line react/destructuring-assignment
-          className: props?.class || '',
-        },
-        children,
+      styled: (children, props) => (
+        <span className={props.class}>{children}</span>
       ),
       bold: (children) => <strong>{children}</strong>,
       italic: (children) => <em>{children}</em>,
+      link: (children, props) => {
+        const {
+          href,
+          target,
+          linktype,
+          anchor,
+        } = props;
+        // Structure the link data so it takes the same shape as sbLink
+        const sbLink = {
+          linktype,
+          cached_url: linktype !== 'email' ? href : '',
+          email: linktype === 'email' ? href : '',
+          anchor,
+          // The WYSIWYG link adds a target="_self" by default which is unnecessary
+          target: target === '_blank' ? '_blank' : undefined,
+        };
+
+        return (
+          <CtaLink
+            sbLink={sbLink}
+            variant={isLightText ? 'inlineDark' : 'inline'}
+            className="children:su-inline"
+            // Custom link attributes are not supported by the rich text renderer currently
+            // Adding rel="noopener" for all eternal links for security reasons
+            rel={linktype === 'url' ? 'noopener' : undefined}
+          >
+            {children}
+          </CtaLink>
+        );
+      },
     },
     nodeResolvers: {
       heading: (children, props) => {
         const { level } = props;
-        if (level > 1 && level < 6) {
-          return (
-            <Heading as={`h${level}`} font="serif">
-              {children}
-            </Heading>
-          );
-        }
+        // This gets you type-1 for h6, type-2 for h5, type-3 for h4 and so on
+        const headingSize = 7 - level;
 
-        if (level === 6) {
-          return (
-            <Heading as="h6" font="serif" size="base">
-              {children}
-            </Heading>
-          );
-        }
-
-        return null;
+        return (
+          <Heading as={`h${level}`} size={headingSize as FontSizeType}>
+            {children}
+          </Heading>
+        );
       },
     },
     blokResolvers: {
-      // TODO: Add components
+      sbCta: (props) => (
+        <SbCta blok={props} />
+      ),
```

Comment:
Added support for inserting the CTA component inside the WYSIWYG. In the future we can add more as needed.